enhancements/kube-apiserver/event-ttl.md

```diff
@@ -0,0 +1,275 @@
+
+#### Impact of configuring 5m TTL
+
+ After filling etcd with approximately 4GB of events over 3 hours, then switching to a 5-minute TTL, we observed a sharp drop in storage usage and memory consumption on etcd after the 3h events have expired and the storage got compacted and defragmented. 
+
+ CPU usage showed a slight initial increase followed by a reduction (measured across both etcd and apiserver components), while apiserver memory remained relatively stable. The compaction duration on etcd demonstrated the expected linear relationship with the number of keys being processed, confirming predictable performance characteristics under this workload.
+
+There were no long-term increases in CPU/memory usage found after configuring a 5m TTL vs. the existing default. 
+
+### Risks and Mitigations
+
+**Risk**: Customers might set the value to 0, which means that the events will never be deleted
+**Mitigation**: The API validation and operator ensures the values are within a reasonable range and never zero (5-180 minutes).
+
+### Drawbacks
+
+- Adds complexity to the configuration API
+- Additional validation and error handling required
+
+## Alternatives (Not Implemented)
+
+1. **Hardcoded Values**: Keep the current fixed value of 3 hours
+   - **Rejected**: Does not meet customer requirements for configurability
+
+2. **Environment Variable**: Use environment variables instead of API configuration
+   - **Rejected**: Less user-friendly and harder to manage
+
+3. **Separate CRD**: Create a separate CRD for event configuration
+   - **Rejected**: Overkill for a single setting, better to include in existing APIServer resource
+
+## Test Plan
+
+The test plan will include:
+
+1. **Unit Tests**: Test the API validation and parsing logic
+2. **E2E Tests**: Test that the event TTL is properly configured on all apiserver after applying the setting
+3. **Performance Tests**: Test the impact of different TTL values on etcd performance
+
+## Tech Preview
+
+The EventTTL feature is controlled by the `EventTTL` feature gate, which is enabled by default in both DevPreview and TechPreview feature sets. This allows the feature to be available for testing and evaluation without requiring additional configuration.
+
+The EventTTL feature gate is implemented in [openshift/api PR #2525](https://github.com/openshift/api/pull/2525) and will be removed when the feature graduates to GA, as the functionality will become a standard part of the platform.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- API is implemented and validated
+- Basic functionality works end-to-end
+- Documentation is available
+- Sufficient test coverage
+- EventTTL feature gate is enabled in DevPreview and TechPreview feature sets
+
+### Tech Preview -> GA
+
+- More comprehensive testing (upgrade, downgrade, scale)
+- Performance testing with various TTL values
```

Comment:
Are there any specific tests we need for GA that we won't already have for tech preview?

I know you have already spent a lot of time manually evaluating the performance concerns and it all seems OK. Are there other manual tests to run, or are you imagining that we will create and run automated benchmarks for this?